modules/app/src/main/java/org/locationtech/jtstest/testbuilder/io/shapefile/PolygonHandler.java

```diff
@@ -299,12 +301,15 @@ public int getShapeType(){
     }
     public int getLength(Geometry geometry){
         
-           MultiPolygon multi;
+          Polygon[] polygonArray;
+          MultiPolygon multi;
         if(geometry instanceof MultiPolygon){
             multi = (MultiPolygon)geometry;
         }
         else{
-            multi = new MultiPolygon(new Polygon[]{(Polygon)geometry},geometry.getPrecisionModel(),geometry.getSRID());
+            polygonArray = new Polygon[]{(Polygon)geometry};
+            geometryFactory = new GeometryFactory(geometry.getPrecisionModel(), geometry.getSRID());
+            multi = geometryFactory.createMultiPolygon(polygonArray);
```

Comment:
Could this be

`multi = geometry.getFactory.createMultiPolygon(polygonArray);`